### DISCUSSION TITLE
[FR] New Download Client: Flood

### DISCUSSION THREAD

**Discussion by mouzzampk** [autobrr/autobrr#1646] [Ideas]
Hi, we already have `rtorrent `but would be great if `flood `can be added too. 

Documentation is here

https://github.com/jesec/flood#integrating-with-flood

https://flood-api.netlify.app/#tag/Torrents

Thanks

**Comment by Igetin** [comment]
Would love to see this implemented, since I'm using Flood as my web UI for rTorrent. I really dislike ruTorrent, and would like to avoid having to set up a separate app or HTTP server solely for autobrr integration.

**Comment by zze0s** [comment]
> Would love to see this implemented, since I'm using Flood as my web UI for rTorrent. I really dislike ruTorrent, and would like to avoid having to set up a separate app or HTTP server solely for autobrr integration.

How are you running Flood? Docker or some other setup? Flood uses the same xmlrpc api to rtorrent as we would but it's proxied via ruTorrent usually. Same with the other clients Flood support, they already have their own web-api's so you can use those directly.

**Comment by Igetin** [comment]
> > Would love to see this implemented, since I'm using Flood as my web UI for rTorrent. I really dislike ruTorrent, and would like to avoid having to set up a separate app or HTTP server solely for autobrr integration.
> 
> How are you running Flood? Docker or some other setup? Flood uses the same xmlrpc api to rtorrent as we would but it's proxied via ruTorrent usually. Same with the other clients Flood support, they already have their own web-api's so you can use those directly.

I have rTorrent and Flood running in their own containers in a Podman pod via Quadlet. I'm using a Unix socket (which is recommended over a TCP port) via rTorrent's `network.scgi.open_local` configuration option. That socket sits on a Podman volume that is mounted to both containers, and I've configured Flood to use that socket for connecting to rTorrent.

…in other words, I guess I would also be fine if autobrr just supported the Unix socket connection method for rTorrent, like Flood does. Then I wouldn't really need Flood support in autobrr. 🙂

**Edit:** Fow now, I've just settled with adding Nginx to the mix, since it can handle the socket on behalf of autobrr.
